core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -494,11 +525,45 @@ public void purgeExpired() {
       }
    }
 
+   /**
+    * This method continues the stage on the CPU executor after the passed in <code>delay</code> stage completes.
+    * This method also prints a message at the trace level with the given <code>traceId</code> to allow for
+    * traceability between thread invocations.
+    * @param delay the delay to wait for before continuing with the returned value.
+    * @param traceId the traceId to allow for code to be followed
+    * @param <V> the return type
+    * @return the stage to return that will continue the execution on the CPU thread
+    */
+   private <V> CompletionStage<V> continueOnCPUExecutor(CompletionStage<V> delay,
```

Comment:
As this is called several times with `CompletableFuture.runAsync` and`CompletableFuture.supplyAsync` we could add convenience methods to reduce verbosity. Something like:

```java
   private <V> CompletionStage<V> supplyOnPersistenceExAndContinue(Supplier<V> supplier, int id) {
      return continueOnCPUExecutor(CompletableFuture.supplyAsync(supplier, persistenceExecutor), id);
   }

   private CompletionStage<Void> runOnPersistenceExAndContinue(Runnable runnable, int id) {
      return continueOnCPUExecutor(CompletableFuture.runAsync(runnable, persistenceExecutor), id);
   }
```

Reply:
Or even replace `int id` with `String traceMessage` and call `getNextTraceNumber` in the convenience methods.

This would mean that `performOnAllTxStores` could just return the Runnable and `commitAllTxStores` could be a two liner:

```java
assert ...
return runOnPersistenceExAndContinue(performOnAllTxStores(predicate, writer -> writer.commit(transaction), "Committing tx for all stores for id %d"));
```

Reply:
Sure, I will take a look at condensing these.

Reply:
So I can change it, but unfortuantely they need to be `IntFunction` and `IntConsumer` which require an additional lambda allocation per op. But the code is more condensed so it seems fine to me.